pkgs/tools/typesetting/tex/texlive/default.nix

```diff
@@ -144,6 +293,72 @@ let
     xzcat "$tlpdb" | sed -rn -f "$tl2nix" | uniq > "$out"
   '';
 
+  # build tlType == "bin" containers based on `binfiles` in TLPDB
+  # see UPGRADING.md for how to keep the list of shebangs up to date
+  # TODO: add extraBuildInputs (?) for adding PATH entries to texlive.combine wrappers
+  mkPkgBin = pname: version: run:
+  { binfiles, scriptsFolder ? pname, postFixup ? "", binaliases ? {}, ... }@args:
+    let binNoAliases = assert (lib.all (e: lib.elem e binfiles) (lib.attrNames binaliases));
+          lib.subtractLists (lib.attrNames binaliases) binfiles;
+    in
+    runCommand "texlive-${pname}.bin-${version}"
+      ( {
+          # metadata for texlive.combine
+          passthru = {
+            inherit pname version;
+            tlType = "bin";
+          };
+          # shebang interpreters
+          # TODO: package SNOBOL4
+          buildInputs = [ bash perl python3 bin.luatex ruby tk ];
```

Comment:
Sad realisation: if I don't substitute, then building `texlive.combined.scheme-infraonly` forces me to compile `bin.core-big`. Not an acceptable behaviour. I'll need to peer at the file extensions in `texlive.tlpdb` and save them in `tlpdb.nix`, in some form (`binfilesExts`??). Then `tests.texlive.shebangs` will verify that we pulled all build inputs successfully.

Reply:
Allegedly fixed.